cypress/integration/unit_tests/click_feature_test.js

```diff
@@ -1,90 +1,201 @@
+      cy.stub(document, 'createElement')
+          .callsFake((tag) => doc.createElement(tag));
+      const containerDiv = doc.createElement('div');
+      containerDiv.id = 'tableContainer';
+      doc.body.appendChild(containerDiv);
+      const tableDiv = doc.createElement('div');
+      tableDiv.id = 'table';
+      containerDiv.appendChild(tableDiv);
+      drawTableAndSetUpHandlers(
+          convertEeObjectToPromise(scoredFeatures).then((fc) => fc.features),
+          map, features);
+    });
+    cy.wrap(loadingFinishedPromise);
+    return assertNoSelection();
+  }
 
-    mockTable.verify();
+  it('clicks on a feature on the map, then unclicks it', () => {
+    setUpPage();
+    clickAndVerifyBlockGroup();
+    cy.wait(waitBeforeClick);
+    cy.get('#test-map-div').click(500, 100);
+    assertNoSelection();
   });
 
-  it('clicks on a block group not in list', () => {
-    const tableData = [tableHeadings, [1, 99, 0.46, 0.52]];
-    mockTable.expects('setSelection').once().withArgs([]);
+  it('clicks on a feature on the map, then clicks on another', () => {
+    setUpPage();
+    clickAndVerifyBlockGroup();
+    cy.wait(waitBeforeClick);
+    cy.get('#test-map-div').click(800, 200);
+    cy.get('#test-map-div').should('contain', 'SCORE: 3');
+    cy.get('#test-map-div').should('contain', 'another group');
+    assertFeatureShownOnMap(feature2Corners);
+  });
 
-    clickFeature(0, 0, null, 'mockAsset', tableApi, tableData);
+  it('click highlights correct feature even after resort', () => {
+    setUpPage();
+    // Sort descending by damage percentage
+    cy.get('.google-visualization-table-tr-head > :nth-child(4)').click();
+    cy.get('.google-visualization-table-tr-head > :nth-child(4)').click();
+    clickAndVerifyBlockGroup();
+  });
 
-    mockTable.verify();
+  it('clicks a place where there is no damage -> no feature', () => {
+    setUpPage();
+    cy.get('#test-map-div').click(900, 100);
+    assertNoSelection();
   });
 
-  it('click then unclick a block group', () => {
-    const tableData = [tableHeadings, [0, 99, 0.46, 0.52]];
-    mockTable.expects('setSelection').once().withArgs([{row: 0, column: null}]);
-    mockTable.expects('setSelection').once().withArgs([]);
+  /**
+   * Asserts that there is currently no pop-up on the map or selected row in the
+   * table.
+   * @return {Cypress.Chainable}
+   */
+  function assertNoSelection() {
+    cy.get('#test-map-div').should('not.contain', 'SCORE:');
+    getDataFeatures().then((features) => expect(features).to.be.empty);
+    return cy.get('.google-visualization-table-tr-sel').should('not.exist');
+  }
 
-    clickFeature(0, 0, null, 'mockAsset', tableApi, tableData);
-    clickFeature(0, 0, null, 'mockAsset', tableApi, tableData);
+  /**
+   * Gets all features currently displayed on map.
+   * @return {Cypress.Chainable<Array<google.maps.Data.Feature>>}
+   */
+  function getDataFeatures() {
+    return cy.wrap(null).then(() => {
```

Comment:
what does the cy.wrap(null) do again? mind leaving a comment?

Reply:
This is a fundamental enough concept that I think this isn't the best place for it. Maybe a README somewhere, though I don't think we have it yet. (I do this a lot more in the follow-up.) Any suggestions for a central place? Or I could create a library function "executeDuringCypressPhase()" that takes a callback, and document there. How about that?

cy.wrap(null).then() makes the code inside the then() only execute in the "Cypress phase" of execution, not during the initial phase. Without the .then(), we'd try to access the map variable even before it was created (because the createMap() function doesn't actually create the map until the Cypress phase).

Reply:
Added a function in commands.js. Will migrate downstream PRs to use it.